session/tokenizer.go

```diff
@@ -32,35 +31,31 @@ type (
 		x.TracingProvider
 		x.HTTPClientProvider
 		config.Provider
-		x.JWKFetchProvider
+		x.JWKSFetchProvider
 	}
 	Tokenizer struct {
 		r       tokenizerDependencies
 		nowFunc func() time.Time
+		cache   *ristretto.Cache
 	}
 	TokenizerProvider interface {
 		SessionTokenizer() *Tokenizer
 	}
 )
 
 func NewTokenizer(r tokenizerDependencies) *Tokenizer {
-	return &Tokenizer{r: r, nowFunc: time.Now}
+	cache, _ := ristretto.NewCache(&ristretto.Config{
+		MaxCost:     50 << 20, // 50MB,
+		NumCounters: 500_000,  // 1kB per snippet -> 50k snippets -> 500k counters
+		BufferItems: 64,
+	})
+	return &Tokenizer{r: r, nowFunc: time.Now, cache: cache}
 }
 
 func (s *Tokenizer) SetNowFunc(t func() time.Time) {
 	s.nowFunc = t
 }
 
-var cache, _ = ristretto.NewCache(&ristretto.Config{
-	NumCounters:        100000000,
-	MaxCost:            10000000,
-	BufferItems:        64,
-	IgnoreInternalCost: true,
-	Cost: func(value interface{}) int64 {
```

Comment:
Same here - please use a cost function and follow best practices for setting counters / cost functions